Source/WebCore/xml/parser/XMLDocumentParser.h

```diff
@@ -190,6 +191,8 @@ class XMLDocumentParser final : public ScriptableDocumentParser, public PendingS
 xmlDocPtr xmlDocPtrForString(CachedResourceLoader&, const String& source, const String& url);
 #endif
 
-std::optional<HashMap<String, String>> parseAttributes(const String&);
+xmlParserInputPtr entityLoader(const char* url, const char* id, xmlParserCtxtPtr);
```

Comment:
`WebCore::entityLoader` is a very generic name. I would move this into the class scope and make it static, so it can be `WebCore::XMLDocumentParser::entityLoader` instead.

Reply:
This function needs to keep its C-linkage since it's called as a function pointer from libxml2.

Can we change it to `WebCore::externalEntityLoader`?

I don't know whether it will keep its C-linkage as a C++ static class method.

Reply:
Are you sure C linkage actually matters? We pass C++ lambda functions to C libraries all the time; the only restriction is no variables can be captured (as then it wouldn't cast to a function pointer). I don't think it can have C linkage if it's part of the WebCore namespace anyway; surely you would need to move it outside the namespace?

But anyway, this was a nitpick suggestion. It doesn't really matter. Feel free to do whatever you prefer.